src/poetry/core/version/pep440/version.py

```diff
@@ -142,6 +142,29 @@ def non_semver_parts(self) -> tuple[int, ...]:
         assert isinstance(self.release.extra, tuple)
         return self.release.extra
 
+    @property
+    def norm_string(self) -> str:
```

Comment:
```suggestion
    def normalize(self) -> str:
```

This might be better served as a method.

Reply:
Although I would suggest we deprecate the `short` parameter in favour of a new parameter `normalize`, and implement the normalisation logic there. We should really just conform to PEP440 as this is the standard we adhere to. 

Since `rev` exists mostly to support legacy package versions, I'd say we support reading `-` but we always write to PEP440 format in core.

Reply:
I changed the property to a method.

I'm not convinced of introducing a new parameter because it makes `to_string()` more complicated and I think everything else than normalization is not really useful.

In a second commit, I changed `to_string()` to always return a normalized string according to PEP 440. I deprecated `short `because it has no effect anymore.

I think we can either go with both commits or only with the first one. What do you think about it?

Reply:
I think we are going in the right direction. The caveat here seems that some projects might loose readability when poetry cli is used to update version. For example, in a case where the project had `version = "1.0.0.alpha1"`, will be bumped to `1.0.0.a1`. Personally, I would have liked to retain the expanded version if it was used as the original here is more human friendly.

Reply:
On the one hand it may be a bit more readable, but on the other hand it is a bit confusing that `version` is not PEP 440 normalized. I have wondered about this myself.

In long term, we will change it anyway, won't we? [PEP 621](https://peps.python.org/pep-0621/#version) says _"Users SHOULD prefer to specify already-normalized versions."_ It's only "should" not "must", but I see an advantage in sticking to a form that people are used to see everywhere else. So if we change it anyway, I'd question if we have to put effort in keeping two variants for some time?

By the way, it would be `1.0.0a2`. (Dots are only for `post` and `dev`, not for pre releases.)